accepted/future-releases/variance/feature-specification.md

```diff
@@ -0,0 +1,373 @@
+      covariant, and
+      the covariant occurrences of `T` in `Ti`
+      for `i` in `0, ..., k` where the `i`th type parameter of `C` is
+      contravariant,
+  - if `S` is `U Function<X0 extends B0, ...., Xk extends Bk>(T0 x0, ...., Tn xn, [Tn+1 xn+1, ..., Tm xm])`,
+      the union of:
+    - the contravariant occurrences of `T` in `U`
+    - the covariant occurrences of `T` in `Ti` for `i` in `0, ..., m`
+  - if `S` is `U Function<X0 extends B0, ...., Xk extends Bk>(T0 x0, ...., Tn xn, {Tn+1 xn+1, ..., Tm xm})`
+      the union of:
+    - the contravariant occurrences of `T` in `U`
+    - the covariant occurrences of `T` in `Ti` for `i` in `0, ..., m`
+
+The invariant occurrences of a type `T` in another type `S` are:
+  - if `S` is `Future<U>`
+    - the invariant occurrences of `T` in `U`
+  - if `S` is `FutureOr<U>`
+    - the invariant occurrencs of `T` in `U`
+  - if `S` is an interface type `C<T0, ..., Tk>`
+    - the union of the invariant occurrences of `T` in `Ti`
+      for `i` in `0, ..., k` where the `i`th type parameter of `C` is
+      covariant or contravariant, and
+      all occurrences of `T` in `Ti`
+      for `i` in `0, ..., k` where the `i`th type parameter of `C` is
+      invariant,
+  - if `S` is `U Function<X0 extends B0, ...., Xk extends Bk>(T0 x0, ...., Tn xn, [Tn+1 xn+1, ..., Tm xm])`,
+      the union of:
+    - the invariant occurrences of `T` in `U`
+    - the invariant occurrences of `T` in `Ti` for `i` in `0, ..., m`
+    - all occurrences of `T` in `Bi` for `i` in `0, ..., k`
+  - if `S` is `U Function<X0 extends B0, ...., Xk extends Bk>(T0 x0, ...., Tn xn, {Tn+1 xn+1, ..., Tm xm})`
+      the union of:
+    - the invariant occurrences of `T` in `U`
+    - the invariant occurrences of `T` in `Ti` for `i` in `0, ..., m`
+    - all occurrences of `T` in `Bi` for `i` in `0, ..., k`
+
+It is a compile-time error if a variance modifier is specified for a type
+parameter declared by a static extension, a generic function type, a
+generic function or method, or a type alias.
+
+*Variance is not relevant to static extensions, because there is no notion
+of subsumption. Each usage will be a single call site, and the value of
+every type argument associated with an extension method invocation is
+statically known at the call site. Similar reasons apply for functions and
+function types. Finally, the variance of a type parameter declared by a
+type alias is determined by the usage of that type parameter in the body of
+the type alias.*
+
+We say that a type parameter _X_ of a type alias _F_ _is covariant_ if it
+only occurs covariantly in the body of _F_; that it _is contravariant_ if
+it occurs contravariantly in the body of _F_ and does not occur covariantly
+or invariantly; that it _is invariant_ if it occurs invariantly in the body
+of _F_ (*with no constraints on other occurrences*), or if it occurs both
+covariantly and contravariantly.
+
+*In particular, an unused type parameter is considered covariant.*
```

Comment:
What does this mean?  Usually I think of variance as specifying the subtyping relation.  But given `typedef F<X> = int`, I expect that `F<num> <: F<double>` and `F<double> <: F<num>`, so in what sense is `X` covariant?

Reply:
Thinking about this again, I noticed that we don't have any equivalence of this kind. The unused type variable is handled in the few situations where that is needed, because a rule applies to, e.g., any non-contravariant type variable, and that includes both unused, invariant, and covariant ones.